CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ add_subdirectory(src)
 
 add_executable(main main.cpp)
 target_link_libraries(main reffine Arrow::arrow_shared)
+target_include_directories(reffine SYSTEM PUBLIC ${CUDA_INCLUDE_DIRS})
```

Comment:
can we make this optional?

Reply:
cmake has a way to enable things optionally depending on certain flags you pass. so we should have a flag to enable CUDA support.